tests/test_recipes.py

```diff
@@ -65,11 +112,50 @@ def test_prune_recipe(recipe_fixture, execute_recipe, nkeep):
 @pytest.mark.parametrize("cache_inputs", [True, False])
 @pytest.mark.parametrize("copy_input_to_local_file", [True, False])
 def test_recipe_caching_copying(
-    netCDFtoZarr_sequential_recipe, execute_recipe, cache_inputs, copy_input_to_local_file
+    netCDFtoZarr_recipe, execute_recipe, cache_inputs, copy_input_to_local_file
 ):
-    """The basic recipe test. Use this as a template for other tests."""
+    """Test that caching and copying to local file work."""
+
+    RecipeClass, file_pattern, kwargs, ds_expected, target = netCDFtoZarr_recipe
+
+    if not cache_inputs:
+        kwargs.pop("input_cache")  # make sure recipe doesn't require input_cache
+    rec = RecipeClass(
+        file_pattern,
+        **kwargs,
+        cache_inputs=cache_inputs,
+        copy_input_to_local_file=copy_input_to_local_file
+    )
+    execute_recipe(rec)
+    ds_actual = xr.open_zarr(target.get_mapper()).load()
+    xr.testing.assert_identical(ds_actual, ds_expected)
+
+
+@pytest.mark.parametrize("cache_inputs", [True, False])
+@pytest.mark.parametrize("copy_input_to_local_file", [True, False])
+def test_recipe_http_caching_copying(
+    netCDFtoZarr_http_recipe, execute_recipe, cache_inputs, copy_input_to_local_file
+):
+    """Test that caching and copying from http to local file work."""
+
+    RecipeClass, file_pattern, kwargs, ds_expected, target = netCDFtoZarr_http_recipe
+
+    if len(file_pattern.merge_dims) != 0:
+        pytest.skip(
+            "This test ensures auth kwargs are correctly passed to the recipe class."
+            "`netcdf_http_file_pattern` fixture currently does not return a properly"
+            " formatted `path_format` for recipes with a `MergeDim.`"
+        )
```

Comment:
Is this something we can fix in the future?

Reply:
Yep! Fixed by https://github.com/pangeo-forge/pangeo-forge-recipes/pull/167/commits/ba9f7904fd79fd74dcf8f0bf8e62c094efe3af3c